include/roboteam_ai/world/Field.h

```diff
@@ -211,6 +271,10 @@ class Field {
     double getRightmostX() const;
     double getBottommostY() const;
     double getTopmostY() const;
+    double getLeftPenaltyX() const;
+    double getRightPenaltyX() const;
+    double getPenaltyTopY() const;
+    double getPenaltyBottomY() const;
```

Comment:
Guess it would be even better if these functions just return an optional. Skills relying on such a value needs to have some fail safe anyway

Reply:
I decided to not document these methods, since my first approach was to have 1 function that return all possible FieldLines, 1 function that returns all possible FieldValues and 1 function that returns all locations. Since nobody liked this design choice, I use multiple getters instead. However I refuse to write documentation for all of them, because this was also one of the reasons why I prefer to have just a single getter for everything instead.

Reply:
Well yeah the documentation isn't really the issue, im just wondering what happens to the actual optional, does it just return .value() or does it return a default value such as 0 if empty()?

Reply:
> Guess it would be even better if these functions just return an optional. Skills relying on such a value needs to have some fail safe anyway

this.

Reply:
It returns a default value 0 if it is undefined which is also what it did before.